RuleTool.py

```diff
@@ -34,91 +34,68 @@ def extract_sentence_cloud(prefix: str) -> str:
-            previous_rule["action"] = original_texts[1]
-        else:
-            previous_rule["action"] = None
-        updated_ruleset.append(previous_rule)
-    return updated_ruleset
-
-
-def separate_action_components(previous_ruleset: list) -> list:
-    updated_ruleset = []
-    for previous_rule in previous_ruleset:
-        components = previous_rule["action"]
-        previous_rule["actions_list"] = []
-        if not components:
+def separate_action_components(rule: dict) -> dict:
+    components = rule["action"]
+    rule["actions_list"] = []
+    components = components[2:-2]
+    components = components.split(";")
+    rule["actions_list"] = components
+    return rule
+
+
+def separate_action_subcomponents(rule: dict) -> dict:
+    subcomponents_list = []
+    for action in rule["actions_list"]:
+        actions = action.split(".")
+        if actions[0].find("AddProp") == -1:
             continue
-        components = components[2:-2]
-        components = components.split(";")
-        previous_rule["actions_list"] = components
-        updated_ruleset.append(previous_rule)
-    return updated_ruleset
-
-
-def separate_action_subcomponents(previous_ruleset: list) -> list:
-    updated_ruleset = []
-    for previous_rule in previous_ruleset:
-        subcomponents_list = []
-        for previous_action in previous_rule["actions_list"]:
-            previous_actions = previous_action.split(".")
-            if previous_actions[0].find("AddProp") == -1 and previous_actions[0].find("RemoveProp") == -1:
-                continue
-            current_action = previous_actions[1][:-1].split("=")
-            subcomponents_list.append(current_action)
-        previous_rule["actions_dict"] = subcomponents_list
-        updated_ruleset.append(previous_rule)
-    return updated_ruleset
-
-
-def select_requested_actions(previous_ruleset: list, requested_actions: list) -> list:
-    updated_ruleset = []
-    for previous_rule in previous_ruleset:
-        matches = 0
-        for previous_action in previous_rule["actions_dict"]:
-            for requested_action in requested_actions:
-                if previous_action[0] == requested_action[0] and previous_action[1] == requested_action[1]:
-                    matches += 1
-                    break
-        if matches >= len(requested_actions):
-            updated_ruleset.append(previous_rule)
-    return updated_ruleset
+        current_action = actions[1][:-1].split("=")
+        subcomponents_list.append(current_action)
+    rule["actions_dict"] = subcomponents_list
+    return rule
+
+
+def determine_requested_actions(rule: dict, requested_actions: list) -> bool:
+    matches = 0
+    for action in rule["actions_dict"]:
+        for requested_action in requested_actions:
+            if action[0] == requested_action[0] and action[1] == requested_action[1]:
+                matches += 1
+                break
+    if matches >= len(requested_actions):
+        return True
```

Comment:
Might be better/safer off using dicts for `requested_actions` as well.

Reply:
I want to brainstorm a more clever and less exacting way to figure this out!

Reply:
this can be simplified to `return matches >= len(requested_actions)`